bfvmm/src/entry/entry.cpp

```diff
@@ -28,26 +28,31 @@
 //
 
 #include <bfgsl.h>
-#include <bfdebug.h>
+#include <bfarch.h>
 #include <bfvcpuid.h>
 #include <bfobject.h>
 #include <bfexports.h>
 #include <bfsupport.h>
+#include <bfcallonce.h>
 #include <bfexception.h>
 
 #include <vcpu/vcpu_manager.h>
 #include <debug/debug_ring/debug_ring.h>
 #include <memory_manager/memory_manager.h>
 
-#include <intrinsics.h>
+static bfn::once_flag g_init_flag;
 
-extern "C" int64_t
-private_init(void)
-{ return ENTRY_SUCCESS; }
+void
+WEAK_SYM init()
```

Comment:
so the only changes to this file are:
- delete the unused code from the old hyperkernel/rsdp changes. This reverts back to the version of entry.cpp that we had prior to the hyperkernel v1 which is good because I never liked that approach. This only removed "seems". It doesn't change the control-flow
- Add in overridable init functions. Originally I put this in the vCPU class which worked great but providing support for a hybrid approach got me thinking that full construction of the vCPU might be helpful so that you can create your own vCPU, and get a version of it that is fully constructed so the code was moved to here instead.

Reply:
The important thing to notice here is that the control-flow is identical to the previous version of this file, and no mods were made to common.c in the driver (since the control-flow is the same). Construction and execution all occur on the proper cores.